src/main/java/com/bandwidth/iris/sdk/IrisClient.java

```diff
@@ -1,186 +0,0 @@
-        return executeRequest(put);
-    }
-
-    public void postFile(String uri, File file, String contentType) throws Exception {
-        HttpPost post = new HttpPost(uri);
-        post.setEntity(contentType == null ? new FileEntity(file) : new FileEntity(file, ContentType.parse(contentType)));
-        executeRequest(post);
-    }
-
-    public void putFile(String uri, File file, String contentType) throws Exception {
-        HttpPut put = new HttpPut(uri);
-        put.setEntity(contentType == null ? new FileEntity(file) : new FileEntity(file, ContentType.parse(contentType)));
-        executeRequest(put);
-    }
-
-    public String buildUserModelUri(String uriSuffix, Map<String, Object> query) throws URISyntaxException {
-        URIBuilder builder = new URIBuilder(this.uri);
-        builder.setPath(baseUserUrl + uriSuffix);
-        if (query != null) {
-            for (String key : query.keySet()) {
-                builder.addParameter(key, query.get(key).toString());
-            }
-        }
-        return builder.build().toString();
-    }
-
-    public String buildUserModelUri(String[] tokens) throws URISyntaxException {
-        return buildUserModelUri(StringUtils.join(tokens, "/"), null);
-    }
-
-    public String buildUserModelUri(String[] tokens, Map<String, Object> query) throws URISyntaxException {
-        return buildUserModelUri(StringUtils.join(tokens, "/"), query);
-    }
-
-    public String buildModelUri(String[] tokens, Map<String, Object> query) throws URISyntaxException {
-        URIBuilder builder = new URIBuilder(this.uri);
-        builder.setPath(baseUrl + StringUtils.join(tokens, "/"));
-        if (query != null) {
-            for (String key : query.keySet()) {
-                builder.addParameter(key, query.get(key).toString());
-            }
-        }
-        return builder.build().toString();
-    }
-
-    public String buildModelUri(String[] tokens) throws URISyntaxException {
-        return buildModelUri(tokens, null);
-    }
-
-    protected IrisResponse executeRequest(HttpUriRequest request) throws Exception {
-        HttpResponse response;
-        Map<String, String> headers = new HashMap<String, String>();
-        IrisResponse irisResponse = new IrisResponse();
-        response = httpClient.execute(request);
-        irisResponse.setResponseBody(response.getEntity() != null ? EntityUtils.toString(response.getEntity()) : "");
-        irisResponse.setStatusCode(response.getStatusLine().getStatusCode());
-        for (Header h : response.getHeaders("Location")) {
-            headers.put(h.getName(), h.getValue());
-        }
-        irisResponse.setHeaders(headers);
-
-        if (!irisResponse.isOK()) {
-            throw new IrisClientException(irisResponse.getResponseBody());
-        }
-        return irisResponse;
-    }
-
-    public String getIdFromLocationHeader(String locationHeader) {
-        return locationHeader.substring(locationHeader.lastIndexOf("/") + 1);
```

Comment:
I doubt that happens much, but this would break if the location header ends with a `\`